ksql-engine/src/main/java/io/confluent/ksql/function/udaf/sum/SumUdaf.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.sum;
+
+import io.confluent.ksql.function.udaf.TableUdaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+@UdafDescription(name = "sum",
+    description = "Sums the column values.")
+
+public final class SumUdaf {
+
+  private SumUdaf() {
+  }
+
+
+  @UdafFactory(description = "Sum long values.")
+  public static TableUdaf<Long, Long, Long> sumLong() {
+
+    return getSumImplementation(
+        0L,
+        Long::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  @UdafFactory(description = "Sum int values.")
+  public static TableUdaf<Integer, Integer, Integer> sumInt() {
+
+    return getSumImplementation(
+        0,
+        Integer::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  @UdafFactory(description = "Sum double values.")
+  public static TableUdaf<Double, Double, Double> sumDouble() {
+
+    return getSumImplementation(
+        0.0,
+        Double::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  @UdafFactory(description = "Sum double values in a list into a single double")
```

Comment:
Use SQL types:

```suggestion
  @UdafFactory(description = "Sum DOUBLE values in an ARRAY into a single DOUBLE")
```

Or maybe:

```suggestion
  @UdafFactory(description = "Sum ARRAY of DOUBLEs into a single DOUBLE result")
```